internal/venice-common/src/main/java/com/linkedin/venice/controllerapi/ControllerClient.java

```diff
@@ -1156,6 +1156,16 @@ public MultiStoreInfoResponse getClusterStores(String clusterName) {
     return request(ControllerRoute.GET_STORES_IN_CLUSTER, params, MultiStoreInfoResponse.class);
   }
 
+  /**
+   * This method gets a list of store names that are ready for compaction.
+   * @param clusterName, the name of the cluster to query for compaction eligible stores
+   * @return The list of store names that are ready for compaction.
+   */
+  public MultiStoreInfoResponse getStoresForCompaction(String clusterName) {
+    QueryParams params = newParams().add(CLUSTER, clusterName);
```

Comment:
`ControllerClient` is currently scoped to a `cluster` as part of creation and thus by default if you were to create a request with `newParams()` it already includes `CLUSTER` parameter that the controller was instantiated with.

However, if the intent of this API is to explicitly take a cluster, you will need to update this to just instantiate an empty `QueryParams` and set the `CLUSTER` parameter.

You can check out how `newParams()` is implemented below to realize it adds the `CLUSTER` parameters through common parameters method.

Reply:
Aha! I see that `newParams()` populate `cluster` downstream.

I wrote this code by referring to `getClusterStores()` and `getStoreLargestUsedVersion()` where the API takes `clusterName` but uses `newParams()`. Can you help me understand why/if `getClusterStores()` and getStoreLargestUsedVersion()` and `getStoresForCompaction()` should instantiate `QueryParams` differently? It seems to me they use `clusterName` similarly.

Reply:
meaning we should remove `clusterName` param since VeniceController is already responsible for particular clusters?

Reply:
Nope. Just instantiate the `QueryParams` differently; i.e., 
```
new QueryParams()
   .setCluster(cluster)
   ....
```

instead of using `newParams()`